controller/controller_server/exception_handler.py

```diff
@@ -0,0 +1,39 @@
+from decorator import decorator
+
+import grpc
+
+import controller.array_action.errors as controller_errors
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ValidationException
+
+logger = get_stdout_logger()
+
+status_codes_by_exception = {
+    ValidationException: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.IllegalObjectName: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.SpaceEfficiencyNotSupported: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotExist: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotMatchCapabilities: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolParameterIsMissing: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ExpectedSnapshotButFoundVolumeError: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ObjectNotFoundError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.PermissionDeniedError: grpc.StatusCode.PERMISSION_DENIED,
+    controller_errors.VolumeAlreadyExists: grpc.StatusCode.ALREADY_EXISTS,
+    controller_errors.NotEnoughSpaceInPool: grpc.StatusCode.RESOURCE_EXHAUSTED
+}
+
+
+def handle_common_exceptions(response_type):
+    @decorator
+    def decorated_handle_common_exceptions(controller_method, servicer, request, context):
+        try:
+            return controller_method(servicer, request, context)
+        except Exception as ex:
+            if type(ex) not in status_codes_by_exception.keys():
```

Comment:
are you sure `.keys()` is needed?

Reply:
It's not mandatory, but I like the Python zen:

> 	    Explicit is better than implicit.

Do you want me to remove it?

Reply:
yes.
I too like the zen of Python
> Beautiful is better than ugly.

> Simple is better than complex.

and the official docs agree

Reply:
Done